### PR TITLE
Refactored JSON parser to not use PCRE

### PR DESCRIPTION
**Pull request by olehermanse**

### CODE REVIEW

**Comment by olehermanse** [comment]
@cf-bottom jenkins, with exotics, please.

**Comment by cf-bottom** [comment]
Sure, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/4586/)

(with exotics)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/4586/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-4586/

**Comment by oleorhagen** [comment]
What's wrong with PCRE?

**Comment by olehermanse** [comment]
> What's wrong with PCRE?

@oleorhagen Nothing, it is an unnecessary dependency if you don't use it.

File: libutils/json.c [hunk 1]
```diff
@@ -2500,19 +2529,19 @@ static JsonParseError JsonParseAsObject(
...
             // This will match the key from { foo : 2 } but not { -foo: 2 }
             if (property_name == NULL
-                && StringMatch("^\\w[-\\w]*\\s*:", *data, NULL, NULL))
+                && (colon = unquoted_key_with_colon(*data)) != NULL)
             {
-                char *colon = strchr(*data, ':');
-
                 // Step backwards until we are on the last whitespace.
 
                 // Note that this is safe because the above regex guarantees
```
Comment (vpodzime) [inline libutils/json.c]: There's no regex anymore.

**Pull request merged**